jakecache.js

```diff
@@ -106,46 +106,48 @@ class JakeCache extends PolyfilledEventTarget {
 
     this[_status] = this.UNCACHED
 
-    navigator.serviceWorker.addEventListener('message', event => {
-      switch (event.data.type) {
-        case 'abort':
-          this.dispatchEvent(new CustomEvent('abort'))
-          break
-        case 'idle':
-          this[_status] = this.IDLE
-          break
-        case 'checking':
-          this[_status] = this.CHECKING
-          this.dispatchEvent(new CustomEvent('checking'))
-          break
-        case 'cached':
-          this[_status] = this.IDLE
-          this.dispatchEvent(new CustomEvent('cached'))
-          break
-        case 'downloading':
-          this[_status] = this.DOWNLOADING
-          this.dispatchEvent(new CustomEvent('downloading'))
-          break
-        case 'updateready':
-          this[_status] = this.UPDATEREADY
-          this.dispatchEvent(new CustomEvent('updateready'))
-          break
-        case 'noupdate':
-          this[_status] = this.IDLE
-          this.dispatchEvent(new CustomEvent('noupdate'))
-          break
-        case 'progress':
-          this.dispatchEvent(new ProgressEvent('progress', event.data))
-          break
-        case 'obsolete':
-          this[_status] = this.OBSOLETE
-          this.dispatchEvent(new CustomEvent('obsolete'))
-          break
-        case 'error':
-          this.dispatchEvent(new ErrorEvent('error', event.data))
-          break
-      }
-    })
+    if (this.pathname && 'serviceWorker' in navigator) {
```

Comment:
Woudln't an early return not have made more sense? It also avoid the extra indentation